### PR TITLE
More functions for "ui"

### PR DESCRIPTION
**Pull request by ArnaudFonzam**
that pull request has all the all the work of intesys. accept all new think in other to align with the intesys work 

Paired with https://github.com/informatici/openhospital-core/pull/877.

### CODE REVIEW

File: rsc/settings.properties [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# This file contains Open Hospital settings
+LANGUAGE=en
+SINGLEUSER=no
+AUTOMATICLOT_IN=no
+AUTOMATICLOT_OUT=no
+AUTOMATICLOTWARD_TOWARD=no
+LOTWITHCOST=yes
```
Comment (mwithi) [inline rsc/settings.properties:7]: only .dist have to be pushed

File: src/main/java/org/isf/opd/dto/OpdWithOperatioRowDTO.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package org.isf.opd.dto;
+
+import java.util.List;
+
+import org.isf.operation.dto.OperationRowDTO;
```
Comment (mwithi) [inline src/main/java/org/isf/opd/dto/OpdWithOperatioRowDTO.java:5]: missing license header

File: src/main/java/org/isf/opd/rest/OpdController.java [hunk 3]
```diff
@@ -63,16 +72,30 @@ public class OpdController {
...
+	protected OperationRowMapper opRowMapper;
+	
 	@Autowired
 	protected OpdMapper mapper;
 	
+	@Autowired
+	protected WardMapper mapperWard;
+	
+	@Autowired
+    protected WardBrowserManager wardManager;
```
Comment (mwithi) [inline src/main/java/org/isf/opd/rest/OpdController.java:88]: indentation

File: src/main/java/org/isf/opd/rest/OpdController.java [hunk 4]
```diff
@@ -111,7 +176,7 @@ ResponseEntity<OpdDTO> newOpd(@RequestBody OpdDTO opdDTO) throws OHServiceExcept
 	ResponseEntity<OpdDTO> updateOpd(@PathVariable("code") int code, @RequestBody OpdDTO opdDTO)
 			throws OHServiceException {
 		LOGGER.info("Update opds code: {}", opdDTO.getCode());
-		if (opdManager.getOpdById(code) == null) {	
+		if (opdManager.getOpdById(code) == null ) {
```
Comment (mwithi) [inline src/main/java/org/isf/opd/rest/OpdController.java:179]:
```suggestion
		if (opdManager.getOpdById(code) == null) {	
```

File: src/main/java/org/isf/admission/rest/AdmissionController.java [hunk 5]
```diff
@@ -212,18 +213,29 @@ public ResponseEntity<List<AdmittedPatientDTO>> allAdmittedPatients()
 	@GetMapping(value = "/admissions/admittedPatients", produces = MediaType.APPLICATION_JSON_VALUE)
 	public ResponseEntity<List<AdmittedPatientDTO>> getAdmittedPatients(
 			@RequestParam(name = "searchterms", defaultValue = "", required = false) String searchTerms,
-			@RequestParam(name = "admissionrange", required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime[] admissionRange,
-			@RequestParam(name = "dischargerange", required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime[] dischargeRange)
```
Comment (mwithi) [inline src/main/java/org/isf/admission/rest/AdmissionController.java:216]: this change was made to automatically parse the String parameters sent from the ui, and so avoid all the dates converations below... are you sure you really need the code this way?

File: rsc/settings.properties (hunk 1 above)
Comment (ArnaudFonzam) [reply rsc/settings.properties:7]: okay

File: src/main/java/org/isf/opd/rest/OpdController.java [hunk 6]
```diff
@@ -94,13 +117,55 @@ ResponseEntity<OpdDTO> newOpd(@RequestBody OpdDTO opdDTO) throws OHServiceExcept
...
 
+	/**
+	 * Create a new {@link Opd with operation}.
+	 * @param opdWithOperationRowDTO
+	 * @return opdWithOperationRowDTO
+	 * @throws OHServiceException
+	 */
+	@PostMapping(value = "/opds/rows", produces = MediaType.APPLICATION_JSON_VALUE)
+	ResponseEntity<OpdWithOperatioRowDTO> newOpdWithOperationRow(@RequestBody OpdWithOperatioRowDTO opdWithOperatioRowDTO) throws OHServiceException {
+		LOGGER.info("store Out patient {}");
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/opd/rest/OpdController.java:136]: There is a placeholder in the log message but no object is provided.

**Comment by mwithi** [comment]
> that pull request has all the all the work of intesys. accept all new think in other to align with the intesys work

is this PR linked to another one on "core" or the latter remains unchanged?

File: src/main/java/org/isf/opd/rest/OpdController.java [hunk 7]
```diff
@@ -132,6 +197,61 @@ ResponseEntity<OpdDTO> updateOpd(@PathVariable("code") int code, @RequestBody Op
...
+	/**
+	 * Updates the specified {@link Opd}.
+	 * @param opdDTO
+	 * @return the code of updated {@link Opd}
+	 * @throws OHServiceException
+	 */
+	@PutMapping(value = "/opds/rows/{code}", produces = MediaType.APPLICATION_JSON_VALUE)
+	ResponseEntity<OpdWithOperatioRowDTO> updateOpdWithOperationRow(@PathVariable("code") int code, @RequestBody OpdWithOperatioRowDTO opdWithOperatioRowDTO)
+			throws OHServiceException {
+		LOGGER.info("Update opds code: {}", opdWithOperatioRowDTO.getOpdDTO().getCode());
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/opd/rest/OpdController.java:210]:
```suggestion
		LOGGER.info("Update opds code: {}", code);
```

File: src/main/java/org/isf/opd/dto/OpdWithOperatioRowDTO.java (hunk 2 above)
Comment (ArnaudFonzam) [reply src/main/java/org/isf/opd/dto/OpdWithOperatioRowDTO.java:5]: ok

File: src/main/java/org/isf/opd/rest/OpdController.java (hunk 4 above)
Comment (ArnaudFonzam) [reply src/main/java/org/isf/opd/rest/OpdController.java:179]: ok

File: src/main/java/org/isf/opd/dto/OpdDTO.java [hunk 8]
```diff
@@ -315,9 +315,18 @@ public String getAgeType() {
...
+		this.ageType = ageType;
+	}
+
+	public WardDTO getWard() {
+		return ward;
+	}
+
+	public void setWard(WardDTO ward) {
+		this.ward = ward;
+	}
```
[the remark this answers is not in the document]
Comment (ArnaudFonzam) [reply src/main/java/org/isf/opd/dto/OpdDTO.java:328]: ok

File: rsc/settings.properties (hunk 1 above)
Comment (ArnaudFonzam) [reply rsc/settings.properties]: ok

File: src/main/java/org/isf/opd/dto/OpdWithOperatioRowDTO.java (hunk 2 above)
Comment (ArnaudFonzam) [reply src/main/java/org/isf/opd/dto/OpdWithOperatioRowDTO.java:27]: i have add the licence header

**Pull request closed without merging**